docs/docsite/rst/dev_guide/developing_modules_checklist.rst

```diff
@@ -46,7 +46,7 @@ The complete module metadata specification is here: https://github.com/ansible/p
     * For password / secret arguments no_log=True should be set.
     * Requirements should be documented, using the `requirements=[]` field.
     * Author should be set, with their name and their github id, at the least.
-    * Ensure that you make use of U() for urls, C() for files and options, I() for params, M() for modules.
+    * Ensure that you make use of `U()` for URLs, `I()` for option names, `C()` for files and option values, `M()` for module names.
```

Comment:
+1 to  "option name" and "option value"

So if I'm reading your proposed change correctly that would give the following examples

      Or if not set the environment variable C(ACME_PASSWORD) will be used.
      ...
      Required if I(state=present) # What about this case, when it's a mix of option name & value?
      ...
      Mutually exclusive with I(project_src) and I(files).
      ...
      See also M(win_copy) or M(win_template).
      ...
      See U(https://www.ansible.com/tower) for an overview.

I'm working on https://github.com/ansible/ansible/pull/21021 at the moment to improve this.